### ISSUE TITLE
Underlying Angular infdig error on multiple searches.

### ISSUE THREAD

**Issue by fuksi** [loklak/loklak_webclient#51]
The search url works fine. When you search for a term, the url will be updated according to the term (so then we can share the link).  

![image](https://cloud.githubusercontent.com/assets/6878677/7611021/108ff6ae-f98a-11e4-8c80-86928f293695.png)  

The new statuses are updated, but there a short flickering when loading, caused by ```Error :[$rootScope:infdig]```

Refer to #48 for the original pull request and how to reproduce the error.

**Comment by sudheesh001** [comment]
@fuksi The problem is with the HTML templating somewhere because of which the `$digest()` iterations reach and hit the 10 iterations mark before aborting. Is [this](http://stackoverflow.com/questions/14376879/error-10-digest-iterations-reached-aborting-with-dynamic-sortby-predicate) of any help ?

**Comment by fuksi** [comment]
@sudheesh001 Thanks for the troubleshoot links, I understand a bit more about $digest now. The problem was when a new path is updated, routing is triggered as well. So when we clicked ```Search```, it wasn't ```vm.update``` who do the job, but the template was reloaded before ```statuses```  was even updated. Because it was reloaded with the new params, the result we got is from ```angular.element(document).ready```. My solution came mostly from [here](http://stackoverflow.com/questions/12422611/angularjs-paging-with-location-path-but-no-ngview-reload/12429133#12429133)
